### DISCUSSION TITLE
Using "erasing" in paint editor after saving

### DISCUSSION THREAD

**Discussion by enviro-apps** [hm21/pro_image_editor#219] [Q&A] [answered]
Hello,
When I start the image editor, I can choose from various editing modes like painting, text, cropping, etc.
 After using the painting mode to draw lines, I save the editing and reopen the painting mode from the bottom nav bar inside the ImageEditor again. 
Is there a way to use the eraser in this situation again? The undo function still works, so the session isn't completely reset. But the eraser doesnt recognize the lines anymore.
If this isn't possible out of the box, how could it be achieved?

**Comment by hm21** [comment, marked as answer]
Currently, there is no way to restore the previous functionality. Once you close the painting editor, all paintings are converted to an editor layer, which can be moved and scaled. Additionally, the undo/redo function will no longer work within the painting editor. Undo/redo will only be available for the added paintings in the main editor.

If you want to create your own solution or submit a PR, you have two options:

1. Write reversed logic for the [export function](https://github.com/hm21/pro_image_editor/blob/ea4052de35f6f9631216041d90ba754ccd3140d2/lib/modules/paint_editor/paint_editor.dart#L579-L671) to enable importing.
2. Update the logic for the [eraser](https://github.com/hm21/pro_image_editor/blob/ea4052de35f6f9631216041d90ba754ccd3140d2/lib/modules/paint_editor/widgets/painting_canvas.dart#L107-L112) so that it removes existing layers when they are hit.

**Comment by enviro-apps** [comment]
Thanks for the answer. I will try the first approach.
